Rearrange functions in MatterBaseTest

asirko-soft opened the pull request:
#### Summary

This is just rearrangement of functions from MatterBaseTest according to their functionality. Relevant docstrings were improved as well.

#### Related issues

Resolves https://github.com/project-chip/matter-test-scripts/issues/597
Depends-On: #39586 

#### Testing

CI and local testing

asirko-soft committed:
rearrange functions in MatterBaseTest

asirko-soft committed:
add some docstrings

asirko-soft committed:
rewrite some docstrings and add more docstrings where dosctrings are missing for external metthods

asirko-soft commented on src/python_testing/matter_testing_infrastructure/chip/testing/matter_testing.py:
```diff
@@ -487,160 +487,228 @@ def __init__(self, *args):
-            FileNotFoundError if pipe file is not found. If None takes the value from the CI argument --app-pipe,  arg --app-pipe has his own file exists check.
+            probable_error, probable_file = extract_error_text()
+            test_steps = self.get_defined_test_steps(self.current_test_info.name)
+            test_step = str(test_steps[self.current_step_index-1]
+                            ) if test_steps is not None else 'UNKNOWN - no test steps provided in test script'
```
there's no place for test to fail before the first step

asirko-soft commented on src/python_testing/matter_testing_infrastructure/chip/testing/matter_testing.py:
```diff
@@ -659,80 +727,335 @@
+                    break
+
+            asserts.assert_is_not_none(ending_step_idx, "mark_step_ranges_skipped was provided with invalid ending_step_num")
+            asserts.assert_greater(ending_step_idx, starting_step_idx,
+                                   "mark_step_ranges_skipped was provided with ending_step_num that is before starting_step_num")
```
for skipping single step exists another function

asirko-soft committed:
Merge branch 'master' into matterbasetest_rearrange_functions

asirko-soft committed:
add docstring as suggested

asirko-soft committed:
Revert "add docstring as suggested"

This reverts commit 832f7ea7a0028c01d96983e0ee8a623ad52eab99.

asirko-soft committed:
apply suggestion

cecille commented on src/python_testing/matter_testing_infrastructure/chip/testing/matter_testing.py:
```diff
@@ -351,160 +351,249 @@ def __init__(self, *args):
-        ''' Retrieves the test step list for the given test
+    #
+    # Mobly Test Controller Methods (Framework Interface)
+    #
```
Can you add a note here about how authors can use these? 

```suggestion
    #
    # The test framework defines a set of named methods that can be used to set up or tear down tests.
    # setup_class is called once after class initialization, once per class, before any test_ methods are run.
    # setup_test is called once before each test_ function is run
    # teardown_test is called once after each test_
    # teardown_class is called after the last test_ function in the class is run
    #
    # Test authors may overwrite these methods to assist in performing setup and tear down.
    # Test classes that overwrite these functions should ensure the base functions are called as appropriate. 
    # setup_ methods  should call the super() method at the start
    # teardown_ methods should call the super() method at the end
    #
```

cecille commented on src/python_testing/matter_testing_infrastructure/chip/testing/matter_testing.py:
```diff
@@ -351,160 +351,249 @@ def __init__(self, *args):
-        steps = self.get_defined_test_steps(test)
-        return [TestStep(1, "Run entire test")] if steps is None else steps
+        Initializes cluster mapping, step tracking, and global test state.
+        Called once per test class by the Mobly framework.
```
```suggestion
        Called once per test class by the Mobly framework.
        
        Test authors may overwrite this method in the derived class to perform setup that is common for all tests.
        This function is called only once for the class. To perform setup before each test, use setup_test.
        Test authors that implement steps in this function need to be careful of step handling if there is
        more than one test in the class.
        Test authors that implement this method should ensure super().setup_class() is called before any
        custom setup.

```

cecille commented on src/python_testing/matter_testing_infrastructure/chip/testing/matter_testing.py:
```diff
@@ -351,160 +351,249 @@ def __init__(self, *args):
-        except AttributeError:
-            return None
+    def teardown_class(self):
+        """Final teardown after all tests: log all problems."""
```
```suggestion
        """Final teardown after all tests: log all problems.
            Test authors may overwrite this method in the derived class to perform teardown that is common for all tests
             This function is called only once per class. To perform teardown after each test, use teardown_test.
             Test authors that implement steps in this function need to be careful of step handling if there is
             more than one test in the class.
             Test authors that implement this method should ensure super().teardown_class() is called after any
             custom teardown code.

        """
```

cecille commented on src/python_testing/matter_testing_infrastructure/chip/testing/matter_testing.py:
```diff
@@ -351,160 +351,249 @@ def __init__(self, *args):
 
-            An empty list means the test will always be run.
+        Resets test state, starts timers, and notifies runner hooks.
+        Called before each test method by the Mobly framework.
```
```suggestion
        Called before each test method by the Mobly framework.
        
        Test authors may overwrite this method in the derived class to perform setup that is common for all tests.
        This is called once before each test_ in the class.
        
        Test authors that implement this method should ensure super().setup_test() is called before any custom setup.
```

cecille commented on src/python_testing/matter_testing_infrastructure/chip/testing/matter_testing.py:
```diff
@@ -351,160 +351,249 @@ def __init__(self, *args):
-        pics = self._get_defined_pics(test)
-        return [] if pics is None else pics
+    def on_fail(self, record):
+        """Handle test failure callback from Mobly framework.
```
```suggestion
        """Handle test failure callback from Mobly framework.
        
            This is called by the base framework.
            Tests should not call this directly.
            Tests should not overwrite this method.
```

cecille commented on src/python_testing/matter_testing_infrastructure/chip/testing/matter_testing.py:
```diff
@@ -351,160 +351,249 @@ def __init__(self, *args):
-                 + Step 2: Authorize this key on the remote host: run ssh-copy-id user@ip once, using your password
-                 + Step 3: From now on ssh user@ip will no longer ask for your password
+    def on_pass(self, record):
+        """Handle test success callback from Mobly framework.
```
```suggestion
        """Handle test success callback from Mobly framework.

            This is called by the base framework.
            Tests should not call this directly. 
            Tests should not overwrite this method.
```

cecille commented on src/python_testing/matter_testing_infrastructure/chip/testing/matter_testing.py:
```diff
@@ -351,160 +351,249 @@ def __init__(self, *args):
-        command = json.dumps(command_dict)
-        dut_ip = os.getenv('LINUX_DUT_IP')
+    def on_skip(self, record):
+        """Handle test skip callback from Mobly framework.
```
```suggestion
        """Handle test skip callback from Mobly framework.

            This is called by the base framework.
            Tests should not call this directly. 
            Tests should not overwrite this method.
```

cecille commented on src/python_testing/matter_testing_infrastructure/chip/testing/matter_testing.py:
```diff
@@ -523,80 +612,335 @@ def get_credentials(self, default: Optional[str] = 0) -> str:
+
+    #
+    # Matter Test API - Step Management & Execution
+    #
```
```suggestion
    #
    # These methods are used to mark test progress for the test harness and logs, to help with test
    # debugging, issue creation and log analysis by the test labs. 
```

asirko-soft committed:
Apply suggestions from code review

Co-authored-by: C Freeman <cecille@google.com>

asirko-soft committed:
Merge branch 'master' into matterbasetest_rearrange_functions

asirko-soft committed:
autopep restyling

asirko-soft committed:
Refactor app_pipe to pipe_name in MatterBaseTest and runner.py

Updated variable names from app_pipe to pipe_name for consistency across MatterBaseTest and runner.py. Adjusted error handling to reflect the new naming convention, ensuring clarity in logging and exception messages.

cecille reviewed (approved):
Would you mind creating a follow up issue to document how to use the record_problem etc. functions? We have this in the SDK docs, but it might make sense to put this right into the file.

asirko-soft committed:
Merge branch 'master' into matterbasetest_rearrange_functions

asirko-soft committed:
fix minor merge conflict resolutions errors

asirko-soft committed:
Merge branch 'master' into matterbasetest_rearrange_functions

cecille committed:
Merge branch 'master' into matterbasetest_rearrange_functions

asirko-soft committed:
Merge branch 'master' into matterbasetest_rearrange_functions

asirko-soft committed:
fix some mypy issues introduced during conflcts resolution
